### PR TITLE
Add storage feb21

### PR DESCRIPTION
**Pull request by hyhdtc**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [x] new API version 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [x] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [x] PowerShell
      - [x] CLI

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [x] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
(https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-02-01/storage.json [hunk 1]
```diff
@@ -2065,6 +2065,43 @@
...
+		  "x-ms-enum": {
+		    "name": "ExpirationAction",
+		    "modelAsString": true
+		  },
+		  "default": "Log",
+		  "description": "The SAS expiration action. Can only be Log."
+		}
+      },
+      "required": [
+        "sasExpirationPeriod"
```
Comment (blueww) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2021-02-01/storage.json:2089]: Per my test, it seems ExpirationAction is also required in SasPolicy, or will report error  "Values for request parameters are invalid: keyPolicy.expirationAction."  (The error is also wrong)

**Review by blueww — APPROVED** [review]
Generally good for me.
Raise a comment.

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-02-01/storage.json (hunk 1 above)
Comment (hyhdtc) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-02-01/storage.json]: Make ExpirationAction as required.
SRP fix will be rolled out in next release

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-02-01/storage.json [hunk 2]
```diff
@@ -0,0 +1,4468 @@
...
+        "name"
+      ],
+      "description": "The custom domain assigned to this storage account. This can be set via Update."
+    },
+    "SasPolicy": {
+      "properties": {
+        "sasExpirationPeriod": {
+          "type": "string",
+          "example": "1.15:59:59",
+          "description": "The SAS expiration period, DD.HH:MM:SS."
```
Comment (Juliehzl) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2021-02-01/storage.json:2073]: @hyhdtc could you add examples for your new features including SasPolicy and KeyPolicy usage?

**Comment by Juliehzl** [comment]
Please also change readme.python.md as
https://github.com/Azure/azure-rest-api-specs/pull/13034/files

File: specification/storage/resource-manager/Microsoft.Storage/stable/2021-02-01/storage.json (hunk 2 above)
Comment (hyhdtc) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2021-02-01/storage.json:2073]: Added samples in StorageAccount Create/Update/GetProperties

**Comment by hyhdtc** [comment]
> Please also change readme.python.md as
> https://github.com/Azure/azure-rest-api-specs/pull/13034/files. I did in commit [c0c5fcd](https://github.com/Azure/azure-rest-api-specs/commit/c0c5fcdc64b8e53f9ac5e154ac2e4dc28e35b7f4).

Thanks Zunli
Updated Java&CSharp Readme as well

**Comment by hyhdtc** [comment]
> Hi @hyhdtc, Your PR has some issues. Please fix the CI sequentially by following the order of `Avocado, semantic validation, model validation, breaking change, lintDiff`.
> 
> Task	How to fix	Priority	Support (Microsoft alias)
> Avocado	[Fix-Avocado](https://github.com/Azure/avocado)	High	ruowan
> Semantic validation	[Fix-SemanticValidation-Error](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/ci-fix.md#semantic-validation)	High	raychen, jianyxi
> Model validation	[Fix-ModelValidation-Error](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/ci-fix.md#model-validation)	High	raychen,jianyxi
> LintDiff	[Fix-LintDiff](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/ci-fix.md#linter-validation)	high	jianyxi, ruoxuan
> If you need further help, please feedback via [swagger feedback](https://aka.ms/swaggerfeedback)."

Already fixed

**Comment by Juliehzl** [comment]
Before merge, we have to make sure that 2021-02-01 api version is already deployed to all regions in public clouds and government clouds and e2e test scenarios could work as expected. 
I have such concern because we get an issue https://github.com/Azure/azure-cli/issues/17189 for US gov cloud and China cloud with api version 2021-01-01.

**Review by blueww — APPROVED** [review]

**Comment by hyhdtc** [comment]
> Before merge, we have to make sure that 2021-02-01 api version is already deployed to all regions in public clouds and government clouds and e2e test scenarios could work as expected.
> I have such concern because we get an issue [Azure/azure-cli#17189](https://github.com/Azure/azure-cli/issues/17189) for US gov cloud and China cloud with api version 2021-01-01.

Thanks Zunli. I can confirm that the api-version is already deployed to all regions in public and all national cloud.

**Pull request merged**